dace/transformation/dataflow/hbm_transform.py

```diff
@@ -0,0 +1,198 @@
+from typing import Any, Dict, Iterable, List, Tuple, Union
+
+import networkx
+from dace import dtypes, properties, registry, subsets, symbolic
+from dace.sdfg import utils, graph
+from dace.codegen.targets import fpga
+from dace.transformation import transformation, interstate
+from dace.sdfg import nodes as nd
+from dace import SDFG, SDFGState, memlet
+
+
+@registry.autoregister
+@properties.make_properties
+class HbmTransform(transformation.Transformation):
+    # type=List[Tuple[str, str, str]]
+    update_array_banks = properties.Property(
+        dtype=List,
+        default=[],
+        desc=
+        "List of (arrayname, value for location['memorytype'], value for location['bank']])."
+        "The shape of the array will be updated depending on it's previous placement."
+        "DDR->HBM: Add a dimension, HBM->HBM: Scale first dimension, HBM->DDR: Remove first dimension."
+    )
+
+    # type=List[str]
+    update_array_access = properties.Property(
```

Comment:
So this is necessary to apply this transformation? It seems like it requires quite a lot of user input at the moment. You need to explain this in more detail for me to understand what's going on.

Is it because we have to apply it to multiple data containers at the same time?

Reply:
Kinda. The point is that for many applications (not axpy) you may have attached data containers that you don't want to be modified by the transformation. If you think of GEMV for example, it probably makes sense to put each of the vectors on 1 bank (or DDR) and distribute the matrix over HBM. This means that you would modify the access to the matrix array (because it now spans accross multiple banks), but not to the vectors

Reply:
I see, but if it's only one container, you could match on an access node. But this might not be possible if you cannot apply the transformation multiple times in a row because of how it manipulates the SDFG